x-pack/plugins/ml/public/application/components/data_grid/common.ts

```diff
@@ -82,6 +83,37 @@ export const getFieldsFromKibanaIndexPattern = (indexPattern: IndexPattern): str
   return indexPatternFields;
 };
 
+/**
+ * Return a map of runtime_mappings for each of the index pattern field provided
+ * to provide in ES search queries
+ * @param indexPatternFields
+ * @param indexPattern
+ * @param clonedRuntimeMappings
+ */
+export const getRuntimeFieldsMapping = (
+  indexPatternFields: string[],
+  indexPattern: IndexPattern,
+  clonedRuntimeMappings?: { [key: string]: RuntimeField }
+) => {
+  if (!Array.isArray(indexPatternFields) || indexPattern === undefined) return {};
+  const ipRuntimeMappings = indexPattern.getComputedFields().runtimeFields;
+  let combinedRuntimeMappings: { [ipField: string]: any } = {};
+
+  if (typeof ipRuntimeMappings === 'object' && Object.keys(ipRuntimeMappings).length > 0) {
+    indexPatternFields.forEach((ipField) => {
+      if (ipRuntimeMappings.hasOwnProperty(ipField)) {
+        combinedRuntimeMappings[ipField] = ipRuntimeMappings[ipField];
+      }
+    });
+  }
+  if (typeof clonedRuntimeMappings === 'object') {
```

Comment:
It might be dangerous to check this way because `typeof null` is also `object`.

Reply:
Updated here https://github.com/elastic/kibana/commit/bc435134b7a10eb60657c077bec7a52659473569